docs/stdlib/json.rst

```diff
@@ -91,39 +93,37 @@ expression <select>` in *JSON mode*, including the type shape.
     ...     filter .name = 'std::bool');
     {'{"name": "std::bool", "timestamp": "2019-04-02"}'}
 
-JSON values can also be cast back into scalars. This casting is
-symmetrical meaning that if a scalar can be cast into JSON, only that
-particular JSON type can be cast back into that scalar:
+JSON values can also be casted back as a scalar type. Casting JSON is
+symmetrical: if a scalar type can be casted back into JSON, only that
+particular JSON type can be casted back into that scalar. Some scalar types
+will have specific conditions:
 
-- JSON *string* can be cast into :eql:type:`str`. Casting
-  :eql:type:`uuid` and :ref:`date and time types
+- JSON *strings* can be casted as a :eql:type:`str` type. Casting
+  :eql:type:`uuid` and :ref:`date/time types
   <ref_std_datetime>` to JSON results in a JSON
-  *string* representing the original value. This means that it is
-  also possible to cast a JSON *string* back into these types. The
-  string value has to be properly formatted (much like in case of
-  a :eql:type:`str` value being cast) or else the cast will raise an
-  exception.
-- JSON *number* can be cast into any of
-  the :ref:`numeric types <ref_std_numeric>`
-- JSON *boolean* can be cast into :eql:type:`bool`
-- JSON *null* is special since it can be cast into an ``{}`` of any type
-- JSON *array* can be cast into any valid EdgeDB array, so it must be
-  homogeneous, and must not contain *null*
-
-A *regular* :eql:type:`tuple` is converted into a JSON *array* when cast
-into :eql:type:`json`, whereas a *named* :eql:type:`tuple` is converted
-into a JSON *object*. These casts are not reversible, i.e. it is not
-possible to cast a JSON value directly into a :eql:type:`tuple`.
+  string representing its original value. This means it is also possible to
+  cast a JSON string back to its original type. The value of the string must
+  also be properly formatted, otherwise an exception will be raised.
+- JSON *numbers* can be casted to any :ref:`numeric type <ref_std_numeric>`.
+- JSON *booleans* can be casted as a :eql:type:`bool`.
+- Nullified JSON is special as it can be casted as an empty set or ``{}`` of
+  :eql:type:`anytype`.
+- JSON *arrays* can be casted to any valid EdgeDB array, as long as it's
+  homogeneous and do not contain *null* as a value.
```

Comment:
Batching these up rather than going line by line with suggestions here.

- "casted" -> "cast"
- standardized around "cast to" rather than "cast as" for consistency
- I removed the emphasis on the type names since they don't indicate a parameter name, and the bullets already show where we are switching JSON "types."
- It's important to retain "those types" in the first bullet while talking about casting back since that is what we're talking about here. It doesn't make sense in this context to generalize.
- I'm not sure that "nullified JSON" is the same as a JSON value `null`. Rather than try to figure that out, I've reverted to what I think is the more common way to refer to that value.
- Stylized `null` as code since it is the value versus the others which are JSON "types".
- "valid EdgeDB array" is encapsulated in the description that follows (homogeneity, lack of `null`, lack of another array), so I let the longer description of EdgeDB array validity stand alone.
- Added lack of null to description of valid array

```rest
- JSON strings can be cast to a :eql:type:`str` type. Casting :eql:type:`uuid` and :ref:`date/time types <ref_std_datetime>` to JSON results in a JSON string representing its original value. This means it is also possible to cast a JSON string back to those types. The value of the UUID or datetime string must be properly formatted to successfully cast from JSON, otherwise EdgeDB will raise an exception.
- JSON numbers can be cast to any :ref:`numeric type <ref_std_numeric>`.
- JSON booleans can be cast to a :eql:type:`bool`.
- JSON ``null`` is unique because it can be cast to an empty set (``{}``) of any type.
- JSON arrays can be cast to an array, as long as it's homogeneous, does not contain ``null`` as an element of the array, and does not contain another array.
```